### PR TITLE
HDDS-8654. Support dynamic changes ozone.readonly.administrators(OM)

### PR DESCRIPTION
**Pull request by z-bb**
## What changes were proposed in this pull request?

Support OM dynamic changes ozone.readonly.administrators.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/projects/HDDS/issues/HDDS-8654

## How was this patch tested?

ut

### CODE REVIEW

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/reconfig/TestOmReconfigure.java [hunk 1]
```diff
@@ -97,4 +98,23 @@ public void testOmAdminUsersReconfigure() throws Exception {
         ozoneManager.getOmAdminUsernames().contains(userB));
   }
 
+  /**
+   * Test reconfigure om "ozone.readonly.administrators"
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/reconfig/TestOmReconfigure.java:102]: Checkstyle complains about missing period:

```
hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/reconfig/TestOmReconfigure.java
 101: First sentence should end with a period.
```

```suggestion
   * Test reconfigure om "ozone.readonly.administrators".
```

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/reconfig/TestOmReconfigure.java [hunk 2]
```diff
@@ -97,4 +98,23 @@ public void testOmAdminUsersReconfigure() throws Exception {
...
+   */
+  @Test
+  public void testOmReadOnlyUSersReconfigure() throws Exception {
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/reconfig/TestOmReconfigure.java:105]: Typo:

```suggestion
  public void testOmReadOnlyUsersReconfigure() throws Exception {
```

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/reconfig/TestOmReconfigure.java [hunk 3]
```diff
@@ -97,4 +98,23 @@ public void testOmAdminUsersReconfigure() throws Exception {
...
+    String userA = "mockUserA";
+    String userB = "mockUserB";
+    conf.set(OZONE_READONLY_ADMINISTRATORS, userA);
+    ozoneManager.reconfigurePropertyImpl(OZONE_READONLY_ADMINISTRATORS, userA);
+    assertTrue(userA + " should be an readOnly admin user",
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/reconfig/TestOmReconfigure.java:110]:
```suggestion
    assertTrue(userA + " should be a readOnly admin user",
```

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/reconfig/TestOmReconfigure.java [hunk 4]
```diff
@@ -97,4 +98,23 @@ public void testOmAdminUsersReconfigure() throws Exception {
...
+        ozoneManager.getOmReadOnlyUsernames().contains(userA));
+
+    conf.set(OZONE_READONLY_ADMINISTRATORS, userB);
+    ozoneManager.reconfigurePropertyImpl(OZONE_READONLY_ADMINISTRATORS, userB);
+    assertFalse(userA + " should NOT be an admin user",
+        ozoneManager.getOmReadOnlyUsernames().contains(userA));
+    assertTrue(userB + " should be an admin user",
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/reconfig/TestOmReconfigure.java:117]:
```suggestion
    assertFalse(userA + " should NOT be a readOnly admin user",
        ozoneManager.getOmReadOnlyUsernames().contains(userA));
    assertTrue(userB + " should be a readOnly admin user",
```

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java [hunk 5]
```diff
@@ -4067,6 +4081,10 @@ public Collection<String> getOmAdminUsernames() {
     return omAdmins.getAdminUsernames();
   }
 
+  public Collection<String> getOmReadOnlyUsernames() {
```
Comment (adoroszlai) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java:4084]: For consistency:

```suggestion
  public Collection<String> getOmReadOnlyAdminUsernames() {
```

(apply change to callers, too)

**Review by adoroszlai — CHANGES_REQUESTED** [review]
Thanks @z-bb for the patch.  Mostly looks good, some very minor style issues noted.

**Comment by z-bb** [comment]
@adoroszlai Thanks for the review, the suggested modification has been completed.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/reconfig/TestOmReconfigure.java [hunk 6]
```diff
@@ -97,4 +98,23 @@ public void testOmAdminUsersReconfigure() throws Exception {
         ozoneManager.getOmAdminUsernames().contains(userB));
   }
 
+  /**
+   * Test reconfigure om "ozone.readonly.administrators."
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/reconfig/TestOmReconfigure.java:102]:
```suggestion
   * Test reconfigure om "ozone.readonly.administrators".
```

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java (hunk 5 above)
Comment (z-bb) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java]: @adoroszlai ok, it's changed now

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/reconfig/TestOmReconfigure.java (hunk 6 above)
Comment (z-bb) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/reconfig/TestOmReconfigure.java]: @adoroszlai ok, it's changed now

**Review by adoroszlai — APPROVED** [review]
Thanks @z-bb for updating the patch.

**Pull request merged**